src/test/java/com/switchfully/parkshark_2024_10/auth/AuthServiceTest.java

```diff
@@ -0,0 +1,102 @@
+package com.switchfully.parkshark_2024_10.auth;
+
+import com.switchfully.parkshark_2024_10.exceptions.UnauthorizedException;
+import com.switchfully.parkshark_2024_10.user.Person;
+import com.switchfully.parkshark_2024_10.user.UserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class AuthServiceTest {
+
+    private AuthService authService;
+    private UserService userService;
```

Comment:
```suggestion
@Autowired
    private AuthService authService;

    @MockitoSpyBean
    private UserService userService;
```

The `Autowired`, as you know, uses Dependency Injection to add the AuthService to this class.
The `MockitoSpyBean` annotation overrides the `UserService` with a _spy_ version of the same Service. That means that the Service will work as before, but with the added bonus that we can verify if and when a method call is done etc.